apps/web/src/stores/documentVersions.ts

```diff
@@ -0,0 +1,19 @@
+import useSWR, { SWRConfiguration } from 'swr'
+
+export default function useDocumentVersions(
+  {
+    commitUuid = 'HEAD',
+    staged = false,
+  }: {
+    commitUuid?: string
+    staged?: boolean
+  },
+  opts?: SWRConfiguration,
+) {
+  const fetcher = (url: string) =>
+    fetch(url, {
+      body: JSON.stringify({ commitId: 'HEAD', staged }),
+    }).then((res) => res.json())
+
+  return useSWR(`/api/commits/${commitUuid}/documents`, fetcher, opts)
```

Comment:
Interesting read https://frontendmasters.com/blog/combining-react-server-components-with-react-query-for-easy-data-management/